components/landingPage/ui/previewHomepage.tsx

```diff
@@ -1,37 +1,28 @@
 'use client'
 
-import { FC, useEffect, useState } from "react";
-import { useSmartLink } from "../../../lib/useSmartLink";
+import { FC, useState } from "react";
 import { parseFlatted, stringifyAsType } from '../../../lib/utils/circularityUtils';
 import { WSL_WebSpotlightRoot } from "../../../models";
-import { KontentSmartLinkEvent } from "@kontent-ai/smart-link";
-import { IRefreshMessageData, IRefreshMessageMetadata } from "@kontent-ai/smart-link/types/lib/IFrameCommunicatorTypes";
+import { applyUpdateOnItemAndLoadLinkedItems } from "@kontent-ai/smart-link";
 import Homepage from "./homepage";
+import { useLivePreview } from "../../../lib/useLivePreview";
 
 type HomepageProps = {
   homepageData: WSL_WebSpotlightRoot
 }
 
 const PreviewHomepage: FC<HomepageProps> = ({homepageData}) => {
   const [homepage, setHomepage] = useState(parseFlatted(stringifyAsType(homepageData)));
-  const sdk = useSmartLink();
 
-  useEffect(() => {
-    const getHomepage = async () => {
-      const response = await fetch(`/api/homepage`);
-      const data = await response.json();
+  useLivePreview(async (data) => {
+    const updatedHomepage = applyUpdateOnItemAndLoadLinkedItems(homepage, data, async codenamesToFetch => {
```

Comment:
It is interesting that this works. I would expect `await` before `applyUpdateOnItemAndLoadLinkedItems`. Does this really work properly?